modules/macros/src/main/scala/qq/droste/macros/impl/Macros.scala

```diff
@@ -33,8 +33,14 @@ object Macros {
     classOrTrait.mods.hasFlag(ABSTRACT) &&
     classOrTrait.mods.hasFlag(SEALED)
 
+    /**
+      * returns wether c extends d
+      */
+    def xtends(c: ClassDef, d: ClassDef): Boolean =
+      c.impl.parents.map(_.toString).exists(_.contains(d.name.toString))
```

Comment:
Should this be stricter than `contains`?

I've been known to be a bit paranoid, but something like this could cause a false positive:

```
@deriveFixedPoint sealed abstract class Prod

object Prod {
  final case class Dummy()
  // ... actual Prod constructors here
}
```

This is because as a `case class`, `Dummy` will automatically extend `Product`, and `"Product.contains("Prod")` would return true.

Reply:
Thanks @ceedubs, good catch!

Reply:
@pepegar did you want to get this addressed and then we can merge?

Reply:
Yeah, I want to address it before merging.